### PR TITLE
this commit will test for getting articles by ID and testing its error handling functions

### PR DESCRIPTION
**Pull request by Hasan-M12**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -47,3 +45,32 @@ describe("GET /api/topics", () => {
...
+        });
+      });
+  });
+  test("400: should respond with a bad message when using an invalid id e.g. /banana", () => {
+    return request(app)
+      .get("/api/articles/banana")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:74]: Good error test here. 

You can also do a test for /api/articles/9999 - you want a 404 not found for that!

File: error/app.error.js [hunk 2]
```diff
@@ -1,3 +1,13 @@
 exports.wrongPathHandler = (req, res, next) => {
   res.status(404).send({ msg: "Incorrect endpoint" });
 };
+
+exports.databaseErrorHandler = (err, req, res, next) => {
+  if ((err.code = "22P02")) {
```
Comment (nc-christian-dutton) [inline error/app.error.js:6]: use strict equals === here

File: models/app.models.js [hunk 3]
```diff
@@ -5,3 +5,14 @@ exports.selectTopics = (req, res) => {
...
+exports.SelectArticlesByID = (id) => {
+  return db
+    .query(
+      `SELECT author, title, article_id, body, topic, created_at, votes, article_img_url FROM articles WHERE article_id = $1`,
+      [id]
+    )
+    .then(({ rows }) => {
+      return rows[0];
+    });
+};
```
Comment (nc-christian-dutton) [inline models/app.models.js:18]: Good job with this so far, if you add the 404 test you may want to check if rows is empty here 👀
